tt_metal/hw/inc/ethernet/dataflow_api.h

```diff
@@ -232,9 +283,53 @@ void eth_send_bytes(
             0, ((num_bytes_sent + src_addr) >> 4), ((num_bytes_sent + dst_addr) >> 4), num_bytes_per_send_word_size);
         num_bytes_sent += num_bytes_per_send;
     }
-    erisc_info->user_buffer_bytes_sent += num_bytes;
+    erisc_info->per_channel_user_bytes_send[0].bytes_sent += num_bytes;
 }
 
+/**
+ * Initiates an asynchronous write from a source address in L1 memory on the local ethernet core to L1 of the connected
+ * remote ethernet core. Also, see \a eth_is_receiver_channel_send_done and \a eth_bytes_are_available_on_channel.
+ *
+ * Non-blocking
+ *
+ * Return value: None
+ *
+ * | Argument                    | Description                                             | Type     | Valid Range | Required |
+ * |-----------------------------|---------------------------------------------------------|----------|-------------|----------|
+ * | src_addr                    | Source address in local eth core L1 memory              | uint32_t | 0..256kB    | True     |
+ * | dst_addr                    | Destination address in remote eth core L1 memory        | uint32_t | 0..256kB    | True     |
+ * | num_bytes                   | Size of data transfer in bytes, must be multiple of 16  | uint32_t | 0..256kB    | True     |
+ * | channel                     | Which transaction channel to use. Corresponds to        | uint32_t | 0..7        | True     |
+ * |                             | per_channel_user_bytes_send in erisc_info_t             |          |             |          |
+ * | num_bytes_per_send          | Number of bytes to send per packet                      | uint32_t | 16..1MB     | False    |
+*  | num_bytes_per_send_word_size| num_bytes_per_send shifted right 4                      | uint32_t | 1..256kB    | False    |
+ */
+FORCE_INLINE
+void eth_send_bytes_over_channel(
+    uint32_t src_addr,
+    uint32_t dst_addr,
+    uint32_t num_bytes,
+    uint32_t channel,
+    uint32_t num_bytes_per_send = 16,
+    uint32_t num_bytes_per_send_word_size = 1) {
+    // assert(channel < 4);
+    uint32_t num_bytes_sent = 0;
+    while (num_bytes_sent < num_bytes) {
+        internal_::eth_send_packet(
+            0, ((num_bytes_sent + src_addr) >> 4), ((num_bytes_sent + dst_addr) >> 4), num_bytes_per_send_word_size);
+        num_bytes_sent += num_bytes_per_send;
+    }
+    erisc_info->per_channel_user_bytes_send[channel].bytes_sent = num_bytes;
+    erisc_info->per_channel_user_bytes_send[channel].receiver_ack = 0;
+    uint32_t addr = ((uint32_t)(&(erisc_info->per_channel_user_bytes_send[channel].bytes_sent))) >> 4;
```

Comment:
user api + ethernet api cleanup, should we just ask users to pass in 16B addresses?

Reply:
That would be ideal, yes. I think this goes into the wider API cleanup discussion (e.g. we could add type info to the alignment requirements so that we can enable certain debug modes more easily if desired to catch non-aligned addrs).